### PR TITLE
[PATCH v3] api: time: clarify packet IO time and timer tick

### PR DESCRIPTION
**Pull request by psavol**
Clarify time specification of packet IO time and timer tick

### CODE REVIEW

**Review by MatiasElo — APPROVED** [review]
Two minor wording suggestions.

File: include/odp/api/spec/packet.h [hunk 1]
```diff
@@ -1847,17 +1847,18 @@ void odp_packet_flow_hash_set(odp_packet_t pkt, uint32_t flow_hash);
...
- * timestamp. Packet input interface timestamp resolution can be checked with
- * odp_pktio_ts_res().
+ * Returns timestamp value recorded into the packet. Use odp_packet_has_ts() to check if the packet
+ * has a valid timestamp. Packet input uses packet IO interface specific time source and thus
+ * timestamp (or nanosecond) values from one interface cannot be mixed with values from another
+ * interface (or time source in general). Packet IO interface timestamp resolution can be checked
+ * with odp_pktio_ts_res() and current time with odp_pktio_time().
+ *
+ * Time API operations (e.g. odp_time_diff()) can be used on packet timestamp values or
+ * in conversion of those into nanoseconds (odp_time_to_ns()).
```
[the remark this answers is not in the document]
Comment (psavol) [reply include/odp/api/spec/packet.h:1857]: Fixed in v2.

File: include/odp/api/spec/timer.h [hunk 2]
```diff
@@ -174,10 +174,14 @@ uint64_t odp_timer_ns_to_tick(odp_timer_pool_t timer_pool, uint64_t ns);
  * Current tick value
  *
  * Returns the current tick value of the timer pool. Timer tick is an implementation defined unit
- * of time. Timer tick value increments with a constant, timer pool specific frequency. Tick
- * frequency may be equal or higher than the requested timer pool resolution. The frequency can be
- * checked with odp_timer_pool_info(). Tick value increments with implementation specific step
- * sizes. The value will not wrap around in at least 10 years from the ODP instance startup.
+ * of time. Tick and related nanosecond values are timer pool specific. Those may not start from
```
[the remark this answers is not in the document]
Comment (psavol) [reply include/odp/api/spec/timer.h:177]: Fixed in v2.

**Comment by PavanNikhilesh** [comment]
Acked-by: Pavan Nikhilesh <pbhagavatula@marvell.com>

**Comment by nithind1988** [comment]
Acked-by: Nithin Dabilpuram<ndabilpuram@marvell.com>

**Pull request merged**